toolbox/src/DiscreteFilter.cpp

```diff
@@ -54,85 +71,59 @@ unsigned DiscreteFilter::numberOfParameters()
 
 bool DiscreteFilter::parseParameters(BlockInformation* blockInfo)
 {
-    ParameterMetadata paramMD_1lowp_fc(ParameterType::STRUCT_DOUBLE, PARAM_IDX_STRUCT, 1, 1, "Fc");
-    ParameterMetadata paramMD_1lowp_ts(ParameterType::STRUCT_DOUBLE, PARAM_IDX_STRUCT, 1, 1, "Ts");
-    ParameterMetadata paramMD_md_order(
-        ParameterType::STRUCT_INT, PARAM_IDX_STRUCT, 1, 1, "MedianOrder");
-    ParameterMetadata paramMD_flt_type(
-        ParameterType::STRUCT_STRING, PARAM_IDX_STRUCT, 1, 1, "FilterType");
-    ParameterMetadata paramMD_numcoeff(ParameterType::STRUCT_DOUBLE,
-                                       PARAM_IDX_STRUCT,
-                                       1,
-                                       ParameterMetadata::DynamicSize,
-                                       "NumCoeffs");
-    ParameterMetadata paramMD_dencoeff(ParameterType::STRUCT_DOUBLE,
-                                       PARAM_IDX_STRUCT,
-                                       1,
-                                       ParameterMetadata::DynamicSize,
-                                       "DenCoeffs");
-    ParameterMetadata paramMD_init_status(
-        ParameterType::STRUCT_BOOL, PARAM_IDX_STRUCT, 1, 1, "InitStatus");
-    ParameterMetadata paramMD_init_y0(
-        ParameterType::STRUCT_DOUBLE, PARAM_IDX_STRUCT, 1, ParameterMetadata::DynamicSize, "y0");
-
-    ParameterMetadata paramMD_init_u0(
-        ParameterType::STRUCT_DOUBLE, PARAM_IDX_STRUCT, 1, ParameterMetadata::DynamicSize, "u0");
-
-    bool ok = true;
-    ok = ok && blockInfo->addParameterMetadata(paramMD_1lowp_fc);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_1lowp_ts);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_md_order);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_flt_type);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_numcoeff);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_dencoeff);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_init_status);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_init_y0);
-    ok = ok && blockInfo->addParameterMetadata(paramMD_init_u0);
-
-    if (!ok) {
-        wbtError << "Failed to store parameters metadata.";
-        return false;
+    const auto& DynPar = ParameterMetadata::DynamicSize;
+
+    const std::vector<ParameterMetadata> metadata{
+        {ParameterType::STRUCT_DOUBLE, ParamIndex::FilterStruct, 1, 1, "Fc"},
+        {ParameterType::STRUCT_DOUBLE, ParamIndex::FilterStruct, 1, 1, "Ts"},
```

Comment:
`MedianOrder` and `FilterType` have been removed?

Reply:
You caught this well Silvio, I realized it running some test controllers and I already have modifications on a local branch ;)